app/models/service_retire_task.rb

```diff
@@ -30,8 +30,10 @@ def after_request_task_create
   end
 
   def create_retire_subtasks(parent_service)
+    parent_service.direct_service_children.each { |child| create_retire_subtasks(child) }
     parent_service.service_resources.collect do |svc_rsc|
       next unless svc_rsc.resource.present? && svc_rsc.resource.respond_to?(:retire_now)
+      next if svc_rsc.resource.type.blank?
```

Comment:
@d-m-u 
Can we validate that this code gets executed. According to @tinaafitz a child service would get marked as an blank resource type.

Reply:
The top level service itself has a blank type and we're not creating task for it to retire in this method so we need this check.